build/lib/src/builder/build_step_impl.dart

```diff
@@ -54,7 +54,7 @@ class BuildStepImpl implements BuildStep {
       Iterable<AssetId> expectedOutputs,
       this._reader,
       this._writer,
-      @deprecated this._rootPackage,
+      @deprecated this._rootPackage, // ignore: deprecated_member_use
```

Comment:
Unless we are actually planning on cleaning this up? I think its necessary though.

Reply:
I think we probably should clean it up - but it would be a breaking because we now reference `BuildStepImpl` in `build_runner`.

We don't read this field anymore do we?
https://github.com/dart-lang/build/pull/379/files#diff-4544b82ad6d6dd0948c77a13cae96758

Reply:
We reference it in a few places in this file to do checks that you aren't trying to read invalid files etc.

Reply:
In that case I'd say remove the `@deprecated`

Reply:
or maybe we intended to infer it from `this.inputId.package`?

Reply:
I don't think its always the same as `this.inputId.package`, it should be the root package in the graph. I can just remove the `@deprecated`.